lab_2_tokenize_by_bpe/main.py

```diff
@@ -94,6 +217,27 @@ def tokenize_word(
     :param unknown_token: token that signifies unknown sequence
     :return: list of token identifiers
     """
+    if not (
+        isinstance(word, tuple) and
+        isinstance(vocabulary, dict) and
+        isinstance(unknown_token, str) and
+        (isinstance(end_of_word, str) or end_of_word is None)
+    ):
+        return None
+    voc_keys = vocabulary.keys()
+    encoded = []
+    word = ''.join(list(word))
+
+    for index, token in enumerate(voc_keys):
+        while token in word:
+            position = word[:word.find(token)].count(' ')
+            encoded.insert(position, index)
+            word = word.replace(token, ' ', 1)
+
+    for index, element in enumerate(word):
+        if element != ' ':
+            encoded.insert(index, vocabulary['<unk>'])
```

Comment:
unknown token is passed as an arguments and it could be something different than `<unk>`

Reply:
fixed